src/constants/LocalStorageCard/LocalStorageCard.ts

```diff
@@ -34,4 +34,53 @@ export const updateUsersInLocalStorage = (getUsers:any, updatedUser:any) => {
         ];
         localStorage.setItem(LOCAL_STORAGE_KEYS.Users, JSON.stringify(updatedUsers));
     }
+};
+
+export const updateAuthMeHistory = (searchHistoryEntry: any) => {
+    const authMe = getAuthMe();
+    if (authMe) {
+        const updatedAuthMe = {
+            ...authMe,
+            History: [...(authMe.History || []), searchHistoryEntry],
+        };
+        localStorage.setItem(LOCAL_STORAGE_KEYS.AuthMe, JSON.stringify(updatedAuthMe));
+        return updatedAuthMe;
+    }
+    return null;
+};
+
+export const updateUsersHistory = (updatedAuthMe: any) => {
+    if (updatedAuthMe) {
+        const users = getUsers();
+        if (users) {
+            const userIndex = users.findIndex((user: any) => user.login === updatedAuthMe.login);
+            if (userIndex !== -1) {
+                const updatedUser = {
```

Comment:
```suggestion
            const updatedUsers = [...users];
            updatedUsers[userIndex] = {...updatedUsers[userIndex], History: updatedAuthMe.History };
```
эти две строки заменят всё что до 67, но более понятно